packages/block-library/src/block/index.php

```diff
@@ -13,10 +13,18 @@
  * @return string Rendered HTML of the referenced block.
  */
 function render_block_core_block( $attributes ) {
+	static $seen_refs = array();
+
 	if ( empty( $attributes['ref'] ) ) {
 		return '';
 	}
 
+	if ( in_array( $attributes['ref'], $seen_refs, true ) ) {
+		return 'nope';
```

Comment:
We should change this :)

Reply:
I'm glad you bring that up! I'm thinking that, for the front end, we should always show a disruptive warning as we do in the editor. We should probably honour the right constants or settings, e.g. `WP_DEBUG`.

If a site disables warnings, then we should just `return '';`. Otherwise, we can trigger a PHP warning.

<img width="631" alt="Captura de ecrã 2021-01-22, às 10 15 50" src="https://user-images.githubusercontent.com/150562/105478334-f8927980-5c9a-11eb-99d5-ae71ad554b2f.png">

My only hesitation is that the PHP warning alone doesn't really help the site admin find the offending block. See the screenshot above. The text that reads _[block rendering halted]_ is returned by `render_block_core_block`.

Reply:
(edited: _should_ -> _should **not**_ 😅)

Reply:
Your idea seems valid to me and you can just add a better message like this:
```
trigger_error("Reusable block `$reusable_block->post_title`, tries to render itself and can cause infinite loop...", E_USER_WARNING);
```

with some different wording probably :)

Reply:
Good idea. Pushed in 9c7d4af1914309aff80bfbd5f2cd6ac5652fa4ab.